cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapsBundleBuilder.java

```diff
@@ -0,0 +1,322 @@
+                // who is responsible for ongoing maintenance of and accessing to this gaps in
+                // care report. Add as a setting and optionally read if it's there.
+                .build();
+    }
+
+    private DetectedIssue getDetectedIssue(
+            Patient patient, MeasureReport measureReport, CareGapsStatusCode careGapStatusCode) {
+        return new DetectedIssueBuilder<>(DetectedIssue.class)
+                .withProfile(CARE_GAPS_DETECTED_ISSUE_PROFILE)
+                .withStatus(DetectedIssue.DetectedIssueStatus.FINAL.toString())
+                .withCode(CARE_GAPS_CODES.get("http://terminology.hl7.org/CodeSystem/v3-ActCode/CAREGAP"))
+                .withPatient(Ids.simple(patient))
+                .withEvidenceDetail(Ids.simple(measureReport))
+                .withModifierExtension(new ImmutablePair<>(
+                        CARE_GAPS_GAP_STATUS_EXTENSION,
+                        new CodeableConceptSettings()
+                                .add(
+                                        CARE_GAPS_GAP_STATUS_SYSTEM,
+                                        careGapStatusCode.toString(),
+                                        careGapStatusCode.toDisplayString())))
+                .build();
+    }
+
+    private void populateEvaluatedResources(MeasureReport measureReport, Map<String, Resource> resources) {
+        measureReport.getEvaluatedResource().forEach(evaluatedResource -> {
+            IIdType resourceId = evaluatedResource.getReferenceElement();
+            if (resourceId.getResourceType() == null || resources.containsKey(Ids.simple(resourceId))) {
+                return;
+            }
+
+            Class<? extends IBaseResource> resourceType = fhirContext
+                    .getResourceDefinition(resourceId.getResourceType())
+                    .newInstance()
+                    .getClass();
+            IBaseResource resource = repository.read(resourceType, resourceId);
+
+            if (resource instanceof Resource) {
+                Resource resourceBase = (Resource) resource;
+                resources.put(Ids.simple(resourceId), resourceBase);
+            }
+        });
+    }
+
+    private void populateSDEResources(MeasureReport measureReport, Map<String, Resource> resources) {
+        if (measureReport.hasExtension()) {
+            for (Extension extension : measureReport.getExtension()) {
+                if (extension.hasUrl() && extension.getUrl().equals(MEASUREREPORT_MEASURE_SUPPLEMENTALDATA_EXTENSION)) {
+                    Reference sdeRef = extension.hasValue() && extension.getValue() instanceof Reference
+                            ? (Reference) extension.getValue()
+                            : null;
+                    if (sdeRef != null
+                            && sdeRef.hasReference()
+                            && !sdeRef.getReference().startsWith("#")) {
+                        IdType sdeId = new IdType(sdeRef.getReference());
```

Comment:
Pass in resource type

Reply:
references should already contain this value